assets/urls.py

```diff
@@ -3,12 +3,15 @@
 from assets import views, models
 from RIGS import versioning
 
-from PyRIGS.decorators import permission_required_with_403
+from django.contrib.auth.decorators import login_required
+from django.views.decorators.clickjacking import xframe_options_exempt
+from PyRIGS.decorators import has_oembed, permission_required_with_403
 
 urlpatterns = [
     path('', views.AssetList.as_view(), name='asset_index'),
     path('asset/list/', views.AssetList.as_view(), name='asset_list'),
-    path('asset/id/<str:pk>/', views.AssetDetail.as_view(), name='asset_detail'),
+    # Lazy way to enable the oembed redirect...
+    path('asset/id/<str:pk>/', has_oembed(oembed_view="asset_oembed")(views.AssetDetail.as_view()), name='asset_detail'),
```

Comment:
Why the difference from how oembed is done for RIGS/urls.py? I understand you need a permission for that, but surely you could use the asset_view perm?

Just trying to ensure consistency

Reply:
We are currently (ab)using asset_view to be a 'view *ALL* information about an asset' rather than a be able to view assets perm. Basic users can see assets without permissions, the view_asset perm is unused in favour of 'view_asset_financial' AFAIK. 

Why not add the 'view_asset' to all users and require it? Because that's the same thing as requiring no perm, but with more steps?  

When I open up RIGS to reveal more information to basic users, the behaviour will be consistent in not requiring permissions for either oembed.

Reply:
Surely though the whole point of permissions is that they can be created and revoked? If the permission exists, we should honour it on the backend so that in the future if an admin wants to create a set of users that don't have access and just revoke the permission without checking that it actually occurs, we don't have a breach?

Personally I think we should have this granular approach, but if you have thought about it and see a reason why not, then that is fine and I'll approve it

Reply:
In a general world situation, totally get you. I think in the case of RIGS I think the usecase is sufficiently different to warrant this. In what possible situation might we need such a group? 

I do recognise it is a bit misleading, but intentionally so because the current usecase would be that these permissions are on all users, which is tedious to apply...

Reply:
Perhaps the solution is to disable/remove the view asset perm, if that's possible?

Reply:
That would make sense, though tbh is as much effort as just doing it the same way as in rigs! The only reason I brought it up as well that we can have one way or doing oembeds, rather than two

https://stackoverflow.com/questions/6062655/remove-or-hide-default-permissions-from-django

Reply:
Well like I say once my other PR is also merged it will be unified. The reality is its unlikely that future maintainers will be without support from us, and if they are then they're probably doing a full rewrite anyway :P

Reply:
Happy to merge?

Reply:
Yeah, approving the merge now